tests/unit/credentials/VaultServiceCredentials.js

```diff
@@ -0,0 +1,172 @@
+const { STS } = require('aws-sdk');
+const sinon = require('sinon');
+const assert = require('assert');
+
+const { Logger } = require('werelogs');
+
+const { VaultServiceCredentials, newSTSClientFromVaultAuth } =
+    require('../../../lib/credentials/VaultServiceCredentials');
+
+const extension = 'test-extension';
+const roleArn = 'arn:aws:iam::000000000000:role/test-role';
+
+const credentialsSet1 = {
+    AccessKeyId: 'ABCD1234567890XXXX',
+    SecretAccessKey: 'qscwdvefb1234567890',
+    SessionToken: '83bdcf990f5fe307db8f12ec1994f35f4022949f36b94655cc4474',
+};
+const credentialsSet2 = {
+    AccessKeyId: 'XXXXABCD1234567890',
+    SecretAccessKey: '1234567890qscwdvefb',
+    SessionToken: '1994f35f4022949f36b94655cc447483bdcf990f5fe307db8f12ec',
+};
+
+function assertCredentials(vaultCredentials, expectedCredentials) {
```

Comment:
Null is only passed once and so I think we should remove the if statement here and just have the one set of asserts, maybe a deep equal if it makes sense but I am assuming there is some reason why it wasn't used. ( Usually happens one object has extra properties we don't care about whenever I suggest using deep )

Reply:
as you had mentioned, a deep check would fail because of other properties in the test results.